### ISSUE TITLE
error with BMP in some sizes

### ISSUE THREAD

**Issue by JoanMCD** [ZinggJM/GxEPD2_32#10]
If I try to show BMP in sizes not a full divisor of 8 in horizontal they got mis-printed on the Display

ESP32, Arduino 1.8.5, newest GxEPD2_32, Waveshare 1,54 Display B/W, GxEPD2_32_Spiffs_Example, GDEP015OC1, BMP 
Header size: 40
Bit Depth: 1
Format: 0
Image size: 40x18 (working)
Image size: 34x18 (not working)

```
void drawBitmaps_other()
{
  int8_t x = 100;
  int8_t y = 100;
  drawBitmapFromSpiffs("/b4.bmp", x, y); //working
  delay(500);
  drawBitmapFromSpiffs("/bb4.bmp", x, y); //not working
  delay(500);
}
```

**Comment by ZinggJM** [comment]
Yes, your contribution is missing an important part to ease the task of the developer: the bitmap causing the issue. I am sure you find a solution to provide this file (zip it, post it in Arduino Forum Display, or add it to a temporary repository of your own, ..). I plan to add a fix, and I would like to test with your file.
You can also try if this works with library GxEPD2, this contains a fix for w%8=!0.

**Comment by JoanMCD** [comment]
I did some more tests... w%8==0 is not all. (?!?) I crop from betty_1 again and again and it's working - only in the exact resolution of 34x28 it fails again. (The log looks well, but on the Display I )

Here is the important part :-)
https://www.dropbox.com/s/ye9chxltlyfcv1c/data.zip?dl=0

also added a Photo of the result on th edisplay with the estimated BMP (betty_4_not_working.bmp) added on the right side

**Comment by ZinggJM** [comment]
The padding formula is wrong, most likely also where I copied it from. Try change at line 261:

`      // BMP rows are padded (if needed) to 4-byte boundary
      uint32_t rowSize = (width * depth / 8 + 3) & ~3;
      if (depth < 8) rowSize = ((width * depth + 8 - depth) / 8 + 3) & ~3;
`

**Comment by JoanMCD** [comment]
Hi ZinggJM, thank you for your fast reply. 
Looks better, the bb4.bmp looks well, but a part on the right side is missing (plus pol of battery  -> 2 px cols),  same with betty_4_not_working.bmp.

**Comment by JoanMCD** [comment]
no problem - thank you for your great library. The fast fix also work in GxEPD2_32. But also here is a small drawback - the last line of the display screen is blinking in B / W ... no problem for now ! - the most important thing is, to know that there is an bug (and that it is not in my code ;)

BTW. the SPIFFS is unknown in (my?) installation - I had to add `#include "SPIFFS.h"` and kick the `listFiles()` call. Also the GxEPD2_32_Spiffs_Loader sketch doesn't can download the pictures - I did it manually (Security problem ?).
